src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -81,6 +106,28 @@ public static string ConstructCommandLine(int packetSize, string address, bool i
                 sb.Append(ttl);
             }
 
+            if (fragmentOption != PingFragmentOptions.Default )
+            {
+                if (s_isBSD)
+                {
+                    // The bit is off by default on OSX & FreeBSD
+                    if (fragmentOption == PingFragmentOptions.Dont) {
+                        sb.Append(" -D ");
+                    }
+                }
+                else if (!s_isBusybox.Value)  // busybox implementation does not support fragmentation option.
```

Comment:
Are there possibly other things beyond the "busybox" implementation that don't support this?  What happens if we come across such an implementation... this will cause all pings to fail by default, right?  Is there any risk there?

Reply:
Yes, there is some risk. I added test with primary intent to catch such case. (ping loopback with this on/off)  We should hit this path only if the option was specified e.g. simple p = new Ping() ; p.Send() will not hit this. Also the privileged path with RAW socket will not hit this. 
The risk is contained to set of specific conditions.
Also for example on Alpine, one can choose to install full ping, it just is not the default. So if we come across distribution with some unusual implementation we can revisit the detection or mitigation.

Reply:
> We should hit this path only if the option was specified 

Won't we hit it if any option was specified, not just this one?

Reply:
Yes. The PingOptions only has Ttl and DontFragment. So it is simple ping with no (default) options or options.